### PR TITLE
Kuznetsov Artyom (Laboratory work №1 option №1)

### PR DESCRIPTION
**Pull request by Kuznetsov-Artyom**
A plugin has been implemented that outputs information about user-defined data types, as well as information about members.
Example:
```cpp
struct Point3D {
  double m_x{};
  double m_y{};
  double m_z{};
};

template <typename T> struct NodeList {
  T data{};
  NodeList<T> *next{};
};

template <typename T> class CheckStaticConstTemplate {
  static const T m_member;
};

template <typename T> const T CheckStaticConstTemplate<T>::m_member{};
```
Result:
```
Point3D (struct)
|_ m_x (double|public)
|_ m_y (double|public)
|_ m_z (double|public)

NodeList (struct|template)
|_ data (T|public)
|_ next (NodeList<T> *|public)

CheckStaticConstTemplate (class|template)
|_ m_member (const T|private|static)
```

### CODE REVIEW

**Review by aobolensk — APPROVED** [review]

File: clang/test/lab1/kuznetsov_artyom/test.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  int m_id{};
+
+  // CHECK: Human (struct)
+  struct Human {
+    // CHECK-NEXT: |_ m_age (int|public)
+    int m_age{};
+    // CHECK-NEXT: |_ m_cash (int|public)
+    int m_cash{};
+  };
+};
```
Comment (m-ly4) [inline clang/test/lab1/kuznetsov_artyom/test.cpp:70]: Could you please add also member field of type `Human` into `User` definition?

File: clang/labs/lab1/kuznetsov_artyom/PrintClassMembers.cpp [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+class PrintClassMembersVisitor final
+    : public clang::RecursiveASTVisitor<PrintClassMembersVisitor> {
+public:
+  explicit PrintClassMembersVisitor(clang::ASTContext *context)
+      : m_context(context) {}
+  bool VisitCXXRecordDecl(clang::CXXRecordDecl *declaration) {
+    if (declaration->isStruct() || declaration->isClass()) {
+      outInfoUserType(declaration);
+
+      auto members = declaration->fields();
```
Comment (m-ly4) [inline clang/labs/lab1/kuznetsov_artyom/PrintClassMembers.cpp]: Could you please also print `static` attribute of members?

**Comment by Kuznetsov-Artyom** [comment]
I have updated the description for the commit.

Added:
1) Output of static class members
2) The Human member in the User class
3) Other tests

Other:
1) Special output for a static class member (see the description)

**Review by m-ly4 — APPROVED** [review]
Thank you for applying suggestions.

**Pull request merged**